extensions/ql-vscode/src/databaseFetcher.ts

```diff
@@ -49,7 +50,9 @@ export async function promptImportInternetDatabase(
     return;
   }
 
-  validateHttpsUrl(databaseUrl);
+  if (!ALLOW_HTTP.getValue()) {
```

Comment:
```suggestion
  if (!ALLOW_HTTP_SETTING.getValue()) {
```

Reply:
Actually, on a deeper thought, can you push this check into the `validateHttpsUrl` method? Rename the method to `validateUrl`? This method parses the URL as well as check for https. It's probably better to continue to check that the URL parses regardless of whether or not it is https. 

Something like this maybe:

```js
function validateUrl(databaseUrl: string) {
  let uri;
  try {
    uri = Uri.parse(databaseUrl, true);
  } catch (e) {
    throw new Error(`Invalid url: ${databaseUrl}`);
  }

  if (!ALLOW_HTTP_SETTING.getValue() && uri.scheme !== "https") {
    throw new Error("Must use https for downloading a database.");
  }
}